lib/active_hash/relation.rb

```diff
@@ -19,7 +19,7 @@ def where(query_hash = :chain)
       return ActiveHash::Base::WhereChain.new(self) if query_hash == :chain
 
       self.records_dirty = true unless query_hash.nil? || query_hash.keys.empty?
-      self.query_hash.merge!(query_hash || {})
+      self.query_list.concat(query_hash&.to_a || [])
```

Comment:
so this is the main bug that this PR is addressing.

The merge will drop the existing key.

Reply:
Exactly